### DISCUSSION TITLE
MC支持PythonIDLE

### DISCUSSION THREAD

**Discussion by exef-star** [Meloong-Git/PCL#3424] [讨论]
想着是否可以吧Python的IDLE移植到MC，顺便把Python的tkinter也移植进去

**Comment by zkitefly** [comment]
？

和启动器啥关系

**Comment by allMagicNB** [comment]
你移植它干啥？

**Reply by exef-star** [reply]
是不是可以把基于Pythontkinter的窗口像Windows一样显示，比如说公告版块

**Comment by mc-124** [comment]
不是一个语言，不可能的
再说py tk和pcl2有啥关系

**Reply by exef-star** [reply]
《万能的Python库》

**Comment by mc-124** [comment]
你要移植你就自己做个python的mc启动器去

**Reply by exef-star** [reply]
已经有了

**Comment by SqrtTan** [comment]
什么神奇玩意

**Comment by wuliaodexiaoluo** [comment]
> 想着是否可以吧Python的IDLE移植到MC，顺便把Python的tkinter也移植进去

不可能，Minecraft 基于 Java，PCL 基于 VB.Net 和 WPF
基于 Java 的 Jython已经停止维护了，Microsoft 发行的基于 DotNet 的 IornPython 也停止维护了，强行支持可能会出现不可预测的问题
而且你不觉得 Tkinter 的默认 UI 很丑吗，就算要移也是用 PyQt5，Tkinter支持的功能非常有限

**Reply by mc-124** [reply]
《IornPython》

**Reply by wuliaodexiaoluo** [reply]
成大事者不拘泥于小节也

**Reply by allMagicNB** [reply]
> 成大事者不拘泥于小节也

你说得对，但是细节决定成败

**Reply by SqrtTan** [reply]
PyQT6都出来了（）

**Reply by wuliaodexiaoluo** [reply]
反正要支持也只能 PyQt
tkinter 个人认为真的很丑（参考 dotnet 的 Win Form），并且好像阉割了部分本该有的功能
